ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/api/P2PNetwork.java

```diff
@@ -34,6 +36,14 @@
    */
   Collection<PeerConnection> getPeers();
 
+  /**
+   * Returns a stream of peers that have been discovered on the network. These peers are not
+   * necessarily connected.
+   *
+   * @return A stream of discovered peers on the network.
+   */
+  Stream<DiscoveryPeer> getDiscoveredPeers();
```

Comment:
I'm not sure a `Stream` should be returned as a property.  Streams are single use and have transient state, and mutations are not propagated back into the returning object. Could this be renamed `streamDiscoveredPeers`?  This may need another PR to push this `getFoo` -> `streamFoo` done properly, but the name would then reflect reality.  

Another option is to return the backing collection, but that looks to be an even deeper refactor to do properly.

Reply:
Up to now, we haven't used any special naming for streams.  But we can discuss offline and maybe create a follow-up PR.